Handlers/LocalizationAttributePartHandler.cs

```diff
@@ -0,0 +1,137 @@
+﻿using Nwazet.Commerce.Models;
+using Orchard.ContentManagement;
+using Orchard.ContentManagement.Handlers;
+using Orchard.Environment.Extensions;
+using Orchard.Localization;
+using Orchard.Localization.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nwazet.Commerce.Settings;
+using Nwazet.Commerce.Services;
+using Orchard.Localization.Services;
+using Orchard;
+using Orchard.UI.Notify;
+
+namespace Nwazet.Commerce.Handlers {
+    [OrchardFeature("Nwazet.AttributesLocalizationExtension")]
+    public class LocalizationAttributePartHandler : ContentHandler {
+        private readonly IProductAttributeService _attributeService;
+        private readonly ILocalizationService _localizationService;
+        private readonly IOrchardServices _orchardServices;
+        private readonly IContentManager _contentManager;
+        public LocalizationAttributePartHandler(
+            IProductAttributeService attributeService,
+            ILocalizationService localizationService,
+            IOrchardServices orchardServices,
+            IContentManager contentManager) {
+
+            _attributeService = attributeService;
+            _localizationService = localizationService;
+            _orchardServices = orchardServices;
+            _contentManager = contentManager;
+
+            T = NullLocalizer.Instance;
+        }
+
+        public Localizer T;
+        protected override void GetItemMetadata(GetContentItemMetadataContext context) {
+            var part = context.ContentItem.As<ProductAttributePart>();
+            if (part != null) {
+                var locPart = context.ContentItem.As<LocalizationPart>();
+                if (locPart != null) {
+                    if (locPart.Culture != null && !string.IsNullOrWhiteSpace(locPart.Culture.Culture)) {
+                        context.Metadata.DisplayText += T(" (culture: {0})", locPart.Culture.Culture).Text;
+                    }
+                    else {
+                        context.Metadata.DisplayText += T(" (culture undefined)").Text;
```

Comment:
Let me put it another way: you are making a UI decision in a place where it can't be overridden. In fact, I object to the whole idea of modifying the `DisplayText`. This information is not part of what you'd usually consider the display text for a product attribute. If you think the way the attribute should be displayed should include the culture and a label, do it in a template, so that it can be overridden by a theme.